File: SPA/src/app/_models/movie.ts
export interface Movie {
    id: number;
    name: string;
    description: string;
    genre: string;
    thumbNail: string;
    pgRating?: string;
    duration: string;
    releaseInfo: Date;
    trailerUrl?: string;
}

export interface UnreleasedMovie extends Movie {
    displayImage: string;
    news: string;
}

export interface ReleasedMovie extends Movie {
    rating: number;
}

File: API/Models/Movie.cs
using System;

namespace API.Models
{
    public class Movie
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Genre { get; set; }
        public string ThumbNail { get; set; }
        public string DisplayImage { get; set; }
        public string PgRating { get; set; }
        public string Duration { get; set; }
        public DateTime ReleaseInfo { get; set; }
        public string TrailerUrl { get; set; }
    }
}